src/MongoDB.Driver.Core/Core/Misc/DefaultRandomByteGenerator.cs

```diff
@@ -0,0 +1,33 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Security.Cryptography;
+
+namespace MongoDB.Driver.Core.Misc
+{
+    internal class DefaultRandomByteGenerator : IRandomByteGenerator
```

Comment:
As per conversation with Mark Benvenuto, for AWS auth I switched to random byte generation (with 256 possible values) instead of existing random character generation (with around 100).